### PR TITLE
add service worker navigation preload

### PR DESCRIPTION
**Pull request by SimonSiefke**
Adds service worker navigation preload as requested in https://github.com/mdn/content/pull/13669 by @hamishwillee

To test this change in Chrome:
1. Load the page
2. Delete the cache `await caches.delete('v1')`
3. Reload
4. The `/sw-test/` response should be preloaded

index

### CODE REVIEW

File: sw.js [hunk 1]
```diff
@@ -8,12 +8,21 @@ const putInCache = async (request, response) => {
...
-const cacheFirst = async ({ request, fallbackUrl }) => {
+const cacheFirst = async ({ request, preloadResponsePromise, fallbackUrl }) => {
   // First try to get the resource from the cache
   const responseFromCache = await caches.match(request);
   if (responseFromCache) {
     return responseFromCache;
   }
+
+  // Next try to use the preloaded response, if it's there
+  const preloadResponse = await preloadResponsePromise;
```
Comment (hamishwillee) [inline sw.js:19]: If preload is not enabled the value will be `undefined`. I presume that awaiting on undefined just falls through?

If so, awesome.

**Review by hamishwillee — APPROVED** [review]
Thanks @SimonSiefke - really appreciate this.

It looks good to me, but I'm no Javascript expert (and I don't have merge rights here). @wbamberg can you please review?

File: sw.js (hunk 1 above)
Comment (SimonSiefke) [reply sw.js:19]: Yes! `undefined === await undefined`

**Comment by hamishwillee** [comment]
@wbamberg Can you please review this - I have no merge rights in this repo

**Comment by Rumyra** [comment]
Looks good to me! All merged @SimonSiefke @hamishwillee 👍

**Pull request merged**